astropy/io/ascii/ui.py

```diff
@@ -577,41 +577,33 @@ def _guess(table, read_kwargs, format, fast_reader):
 
             reader = get_reader(**guess_kwargs)
 
-            reader.guessing = True
-            dat = reader.read(table)
```

Comment:
I think the wrong thing was moved to address https://github.com/astropy/astropy/pull/15196#discussion_r1299229427.
You can take a look at the CI log files to help diagnose the issue, but it's probably this.